swatch-utilization/src/main/java/com/redhat/swatch/utilization/service/UtilizationSummaryValidator.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.utilization.service;
+
+import com.redhat.swatch.configuration.registry.MetricId;
+import com.redhat.swatch.configuration.registry.ProductId;
+import com.redhat.swatch.configuration.util.MetricIdUtils;
+import com.redhat.swatch.utilization.model.UtilizationSummary;
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@ApplicationScoped
+public class UtilizationSummaryValidator {
+
+  public boolean isValid(UtilizationSummary payload) {
+    return hasValidOrgId(payload)
+        && hasValidProduct(payload)
+        && hasValidMeasurements(payload)
+        && hasValidBillingForPayg(payload);
+  }
+
+  private boolean hasValidOrgId(UtilizationSummary payload) {
+    if (payload.getOrgId() == null) {
+      log.warn("Received utilization summary without orgId. Payload: {}", payload);
+      return false;
+    }
+
+    return true;
+  }
+
+  private boolean hasValidProduct(UtilizationSummary payload) {
+    try {
+      ProductId.fromString(payload.getProductId());
+    } catch (IllegalArgumentException e) {
+      log.warn(
+          "Received utilization summary with invalid productId '{}'. Payload: {}",
```

Comment:
Should this warning include the expected product id in order to debug if there is ever a mismatch?